src/option.h

```diff
@@ -478,19 +478,56 @@ namespace librealsense
         std::function<void(const option&)> _recording_function = [](const option&) {};
     };
 
+    class proxy_option : public option
```

Comment:
Document the class and its purpose

Reply:
"Option wrapper for another -- forwards all API calls to the proxied option such that specific functionality can be easily overriden"